### DISCUSSION TITLE
[Section 5.3 - Buckets] - Problema ao executar terraform init

### DISCUSSION THREAD

**Discussion by jonnesp** [chgasparoto/curso-aws-com-terraform#150] [Q&A]
Pelo que vi você atualizou seu próprio módulo para suportar versões superiores a do terraform 1.0, e o provider da aws também, com isso ao tentar seguir o passo a passo do curso dificulta muito pois acaba perdendo muito tempo tendo que corrigir erros ou encontrar alternativas, no caso vou criar os resources manualmente.... Mas tentando dar o init a partir do próprio fonte vem com erro conforme imagem a seguir:

![Erro terraform init](https://github.com/chgasparoto/curso-aws-com-terraform/assets/13646224/df654ba2-6850-4edf-bf64-332cb8003b12)

Um exemplo de como ficou dentro do módulo importado:

![modulo importado](https://github.com/chgasparoto/curso-aws-com-terraform/assets/13646224/90416e7e-c718-4789-942e-1a6bd9103900)

Gostaria muito de saber se o motivo dos erros foi realmente as alterações do próprio módulo fornecido para o curso ou se estou equivocado.

Obrigado.

**Comment by chgasparoto** [comment]
@jonnesp você precisa adicionar a versão do módulo. Se você não adicionar ele vai pegar a última versão e a última versão desse meu módulo não está compatível com o que é mostrado no curso atualmente.

Por exemplo:

```hcl
module "website" {
  source        = "github.com/chgasparoto/terraform-s3-object-notification?ref=v1.0.3"
```

Ref:

- https://github.com/chgasparoto/terraform-s3-object-notification/releases
- https://developer.hashicorp.com/terraform/language/modules/sources

Eu estou atualizando o curso, logo logo o código do curso estará compatível novamente.

**Reply by jonnesp** [reply]
Muito Obrigado!
